service_restaurant/src/main/kotlin/org/thechance/service_restaurant/usecase/gateway/MealGateway.kt

```diff
@@ -0,0 +1,16 @@
+package org.thechance.service_restaurant.usecase.gateway
+
+import org.thechance.service_restaurant.entity.Meal
+
+interface MealGateway {
+
+    suspend fun addMeal(meal: Meal): Boolean
+
+    suspend fun getMeals(page: Int , limit : Int): List<Meal>
+
+    suspend fun getMealById(id: String): Meal?
+
+    suspend fun deleteMealById(id: String) : Boolean
```

Comment:
i think it will be better to name it just `deleteMeal` without `ById` Because it is always deleted by id